File: Leetcode/cpp/sol1768.cpp
//https://leetcode.com/problems/merge-strings-alternately/solutions/6583176/100-time-by-antoniojsp-3re4/

class Solution {
public:
    string mergeAlternately(string word1, string word2) {

        string result = "";
        int i = 0, j = 0;
        while((i<word1.length()) || (j<word2.length()) ){

            if (i<word1.length()){
                result+=word1[i];
                i++;

            }
            if(j<word2.length()){
                result+=word2[j];
                j++;
            }
        }

        return result;

    }
};

File: Leetcode/python/sol1768.py
#  https://leetcode.com/problems/merge-strings-alternately/

class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:

        result = []
        i = 0
        j = 0
        while i < len(word1) or i < len(word2):
            if i < len(word1):
                result.append(word1[i])
            if j < len(word2):
                result.append(word2[j])
            i += 1
            j += 1

        # if i < len(word1):
        #     result.append(word1[i:])
        # elif j < len(word2):
        #     result.append(word2[i:])

        return "".join(result)

